src/pages/billing/Project.js

```diff
@@ -716,8 +754,17 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
         setCostPerWorkspace(costPerWorkspace)
         setUpdatingProjectCost(false)
       }
-    })
-    maybeLoadProjectCost()
+    }
+    maybeLoadProjectCost().catch(async error => {
+      if (error instanceof Response && error.status === 401) {
+        if (!await reloadAuthToken()) {
```

Comment:
Can we keep auth related code out of this module and avoid duplicating this code by using one of the standard error handling wrappers?

Reply:
I think `withErrorHandling` is probably what you want. Or I think the `catch` block works if you simply delete the 401-specific code. 

@nawatts were you thinking of something else? I know you often just do try/catch.

Reply:
Looking at it again, the code for handling expired auth tokens is in `reportError` and this doesn't report the error with a notification. That's probably the wrong place for that code, but that's beside the point. I'll retract my objection.

https://github.com/DataBiosphere/terra-ui/blob/f1c195a9e58f40d5f756df794147210d6be86cb6/src/libs/error.js#L6-L16

Reply:
So the point of having the 401 handling is to avoid showing an error about the session timed out inline in the spend report page itself? That's probably a good thing, but if the reload of the Auth token is successfully we would still need to re-execute the Ajax call to load the spend report data, wouldn't we?

In fact, I wonder if this pattern is why I often get an empty terra-ui page when I first bring up Terra-- the auth token had timed out and it refresh the token, but it still doesn't load any of the actual page content.

Reply:
Yeah, I think the existing handling for expired auth tokens is broken in multiple ways:
1. It's in `reportError`. Not every Ajax call is wrapped in `withErrorReporting`. It should be in Ajax.
2. As you point out, it doesn't retry. So whatever made the call doesn't work, but now no error message is shown.

Reply:
I have created https://broadworkbench.atlassian.net/browse/SUP-743 to fix the underlying issue with auth token expiration handling. @yunip20 for now I think it is fine to keep this code in your PR, and it can be cleaned up when there is more complete handling that retries the original Ajax call.